test/prioritized-pools.test.js

```diff
@@ -0,0 +1,261 @@
+        expect(connection.pools[0].connection.connected).toEqual(true);
+        expect(connection.pools[1].connection.connected).toEqual(false);
+        // force a prioritize cycle
+        jest.runOnlyPendingTimers();
+      })
+      // need to wait for the heal to finish during the prioritize cycle
+      .then(delay(1000, realSetTimeout))
+      .then(() => {
+        // confirm priority1 pool is now second and priority0 is now first
+        // and that they are both connected
+        expect(connection.pools[0].dsn.priority).toEqual(0);
+        expect(connection.pools[1].dsn.priority).toEqual(1);
+        expect(connection.pools[0].connection.connected).toEqual(true);
+        expect(connection.pools[1].connection.connected).toEqual(true);
+      });
+  });
+  it('does not reprioritize if all higher priority pools cannot be healed', () => {
+    connection = new sql.ConnectionPoolParty({
+      // the dsns are the same, but are sufficient for these tests
+      dsns: [
+        {
+          user: 'sa',
+          password: 'PoolPartyyy9000',
+          server: 'localhost',
+          database: 'PoolParty',
+          priority: 1,
+        },
+        {
+          user: 'sa',
+          password: 'wrong password',
+          server: 'localhost',
+          database: 'PoolParty',
+          priority: 0,
+        },
+      ],
+      prioritizePools: true,
+      // this helps prevent warmup from resolving before all the pools are created
+      warmupStrategy: serialWarmupStrategy,
+    });
+    return connection.warmup()
+      .then(() => {
+        // confirm priority1 pool is first and priority0 is second
+        expect(connection.pools[0].dsn.priority).toEqual(1);
+        expect(connection.pools[1].dsn.priority).toEqual(0);
+        // confirm priority1 pool is healthy but priority0 is not
+        expect(connection.pools[0].connection.connected).toEqual(true);
+        expect(connection.pools[1].connection.connected).toEqual(false);
+        jest.runOnlyPendingTimers();
+      })
+      // need to wait for the heal to finish during the prioritize cycle
+      .then(delay(1000, realSetTimeout))
+      .then(() => {
+        // confirm priority1 pool is still primary
+        expect(connection.pools[0].dsn.priority).toEqual(1);
+        expect(connection.pools[1].dsn.priority).toEqual(0);
+        expect(connection.pools[0].connection.connected).toEqual(true);
+        expect(connection.pools[1].connection.connected).toEqual(false);
+      });
+  });
+  it('does not heal pools with a lower priority than the primary during a prioritize cycle', () => {
+    connection = new sql.ConnectionPoolParty({
+      // the dsns are the same, but are sufficient for these tests
+      dsns: [
+        {
+          user: 'sa',
+          password: 'PoolPartyyy9000',
+          server: 'localhost',
+          database: 'PoolParty',
+          priority: 1,
+        },
+        {
+          user: 'sa',
+          password: 'PoolPartyyy9000',
```

Comment:
We need another test w/ 3 dsns and wrong password here.  With the current implementation of the sorting ignoring failed healing, it will put a disconnected pool first.